warehouse/legacy/api/simple.py

```diff
@@ -84,8 +86,8 @@ def simple_detail(project, request):
                           .with_entities(Release.version)
             )
         )
-        .order_by(File.filename)
-        .all()
+        .all(),
+        key=lambda f: (pkg_resources.parse_version(f.version), f.packagetype)
```

Comment:
So there is a potential for a problem here, if f.packagetype is None problems occur (is that valid? I can make it happen in the tests but not sure it can happen in a real deployment)

Reply:
And similarly, if f.version is None? Looking at recent changes this happens relatively regularly.

Reply:
`File.version` and `File.packagetype` will not be `None` here, and if they are, we've got bigger problems.

Also, I think I would prefer if we used `packaging.version.parse` here instead, from the `packaging` package.

Reply:
We should be good to go then